File: cursor_generate.js
const fs = require('fs').promises;
const path = require('path');

async function main() {
  if (process.argv.length !== 4) {
    console.error('Usage: node cursor_generate.js <filename> "<code>"');
    process.exit(1);
  }

  const [, , filename, code] = process.argv;
  const outPath = path.resolve(filename);

  try {
    await fs.writeFile(outPath, code, 'utf-8');
    console.log(`[INFO] 코드가 ${outPath} 에 저장되었습니다.`);
  } catch (err) {
    console.error('[ERROR] 파일 저장 중 오류 발생:', err);
    process.exit(1);
  }
}

main();

File: cursor_generate.py
# -*- coding: utf-8 -*-
"""
Cursor가 생성한 코드를 지정된 파일에 저장하는 헬퍼 스크립트
사용법: python cursor_generate.py <파일명> <코드 문자열>
예시: python cursor_generate.py my_module.py "def hello():\n    print('안녕')"
"""

import sys
import os

def main():
    if len(sys.argv) != 3:
        print("Usage: python cursor_generate.py <filename> <code>")
        sys.exit(1)

    filename, code = sys.argv[1], sys.argv[2]
    out_path = os.path.abspath(os.path.join(os.getcwd(), filename))

    # 파일이 이미 존재하면 덮어쓰기 전 확인 (필요 시 수정)
    with open(out_path, "w", encoding="utf-8") as f:
        f.write(code)

    print(f"[INFO] 코드가 저장되었습니다: {out_path}")

if __name__ == "__main__":
    main()
